app/controllers/spree/products_controller_decorator.rb

```diff
@@ -4,6 +4,13 @@ def self.prepended(base)
       base.before_action :can_show_product?, only: :show
     end
 
+    def index
```

Comment:
@tomurb please include changes to this file in `spree/spree` - thanks!

Reply:
@damianlegawiec done https://github.com/spree/spree/pull/10436

Reply:
Although maybe it should be made with base branch which will be later a spree-multi-domain migration branch.